rfcs/document_all_media_properties/20230307-project_proposal.md

```diff
@@ -0,0 +1,133 @@
+work. As such, it will be a work in progress because the links to the code and
+the details about the properties should change as we progress through the data
+normalization project. We can publish it on the monorepo documentation site or
+the Openverse blog on Make WordPress, with the admonition that the table is a
+living document and will be updated.
+
+The table will start with a list of columns in the Catalog database. For each
+property, we will provide the following information:
+
+- A clear definition of what the property represents.
+- An explanation of where the property is sourced from in the catalog, and any
+  special considerations that should be taken into account when extracting it.
+  For instance, in the case of the `creator` field, we will specify whether to
+  use the `author`, `photographer`, or both.
+- The property's data type, and whether it is required or nullable. We will also
+  indicate if the property is required but not present in some data, such as
+  `license_url` in the `meta_data` field. If the property has a set of distinct
+  values, we will note where they are in the code.
+- Whether the property is served by the API, and if so, on which endpoints
+  (search or single result, or both). We will also highlight any differences
+  between the property's type in the API versus the catalog, as noted in the
+  database, Elasticsearch, or API documentation.
+- If the API does not take the data directly from the catalog database, we will
+  indicate how the data is computed, and where this occurs (during data refresh,
+  indexing, or as a fallback in the API). We will provide a link to the relevant
+  code, if available.
+- We will also document whether the property is used by the frontend, whether
+  the type matches the catalog/API response type, and whether the frontend
+  performs any cleanup or sanitization of the property. A link to the relevant
+  code will be provided, if available.
+
+#### Tools
+
+The main tool to use for this table is an online spreadsheet app. I'm a little
+uncertain of whether the amount of information would be too much to fit into one
+table. If it is, it might be a good idea to split the data into a table with
+more standard pieces of information, and a list of all data properties with
+longer descriptions and things like code links and the criteria for selecting
+data from provider APIs.
+
+#### The process for updating the table
+
+In the future, when the data is cleaned or changed in any way, we will need to
+update the table accordingly. If we keep it in the documentation site, we can
+use PRs to keep the history of changes, linking to all the related PRs.
+
+#### Localization concerns
+
+We should identify whether any pieces of data are translated or left as is, and
+whether it might be necessary (and possible) to translate any pieces. There is
+an [open issue](https://github.com/WordPress/openverse/issues/589) about
+translating audio genres, for example.
+
```

Comment:
Other concerns

It will not be possible to document all of the fields inside the meta_data column. It is a JSON column that was used to collect "whatever doesn't fit in the other fields" and its values can vary greatly. It would be extremely time- and resources- consuming to try to extract _all_available meta_data fields. For this column, we will describe what we currently expect as a bare minimum value for it, and also the meta_data fields that are used in the provider scripts

Reply:
Does the data that doesn't fit anywhere else ever get used by the API or front end? I know that this is probably beyond the scope of this proposal, but I feel like unless we know that the API and/or front end are using those fields, we could keep them in S3 or something for a rainy day, but keeping them in postgres just degrades performance. :/

Reply:
As far as I know, we only use the `license_url` and `description` fields from the `meta_data` column. I am not sure how much other data we have in there, so creating this table will be a great opportunity to start exploring :) I don't think I can comment on keeping the data in S3 or the performance degradation until I have more data, but I would love to find out.